meta/autoid/autoid.go

```diff
@@ -91,25 +92,65 @@ func (alloc *allocator) NextGlobalAutoID(tableID int64) (int64, error) {
 		return errors.Trace(err1)
 	})
 	metrics.AutoIDHistogram.WithLabelValues(metrics.GlobalAutoID, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
-	return autoID + 1, errors.Trace(err)
+	if alloc.isUnsigned {
+		return int64(uint64(autoID) + 1), err
+	}
+	return autoID + 1, err
 }
 
-// Rebase implements autoid.Allocator Rebase interface.
-// The requiredBase is the minimum base value after Rebase.
-// The real base may be greater than the required base.
-func (alloc *allocator) Rebase(tableID, requiredBase int64, allocIDs bool) error {
-	if tableID == 0 {
-		return errInvalidTableID.GenWithStack("Invalid tableID")
+func (alloc *allocator) rebase4Unsigned(tableID int64, requiredBase uint64, allocIDs bool) error {
+	// Satisfied by alloc.base, nothing to do.
+	if requiredBase <= uint64(alloc.base) {
+		return nil
+	}
+	// Satisfied by alloc.end, need to update alloc.base.
+	if requiredBase <= uint64(alloc.end) {
+		alloc.base = int64(requiredBase)
+		return nil
+	}
+	var newBase, newEnd uint64
+	startTime := time.Now()
+	err := kv.RunInNewTxn(alloc.store, true, func(txn kv.Transaction) error {
+		m := meta.NewMeta(txn)
+		currentEnd, err1 := m.GetAutoTableID(alloc.dbID, tableID)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		uCurrentEnd := uint64(currentEnd)
+		if allocIDs {
+			newBase = mathutil.MaxUint64(uCurrentEnd, requiredBase)
+			newEnd = mathutil.MinUint64(math.MaxUint64-uint64(step), newBase) + uint64(step)
+		} else {
+			if !alloc.isUnsigned && uCurrentEnd >= requiredBase {
```

Comment:
`if !alloc.isUnsigned` check is redundant, already check in line#203

Reply:
I think check !alloc.isUnsigned is not necessary.